File: Programmers_python/기능개발.py
import math

def solution(progresses, speeds):
    release = []
    answer = []
    
    for i in range(len(progresses)):
        release.append(math.ceil((100-progresses[i]) / speeds[i]))
    
    cnt = 1
    release_max = release[0]
    for i in range(1, len(release)):
        if release_max >= release[i]:
            cnt += 1
        else:
            release_max = release[i]
            answer.append(cnt)
            cnt = 1
            
    answer.append(cnt)    
    return answer

File: Programmers_js/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    let days = [];
    
    for(let i=0,len=speeds.length; i<len; i++) {
        days[i] = Math.ceil((100 - progresses[i]) / speeds[i]);
    }
    
    answer.push(1);
    let max_day = days[0];
    
    for(let i=1, len=days.length; i<len; i++) {
        if(max_day >= days[i]) {
            answer.push(answer.pop() + 1);
        } else {
            answer.push(1);
            max_day = days[i];
        }
    }
    
    return answer;
}
